module/zfs/zfs_ioctl.c

```diff
@@ -5534,31 +5542,64 @@ dump_bytes_cb(void *arg)
 	dbi->dbi_err = zfs_file_write(fp, buf, dbi->dbi_len, NULL);
 }
 
+typedef struct dump_bytes_arg {
+	zfs_file_t	*dba_fp;
+#ifdef USE_SEND_TASKQ
+	taskq_t		*dba_tq;
+#endif
+} dump_bytes_arg_t;
+
 static int
 dump_bytes(objset_t *os, void *buf, int len, void *arg)
 {
+	dump_bytes_arg_t *dba = (dump_bytes_arg_t *)arg;
 	dump_bytes_io_t dbi;
 
-	dbi.dbi_fp = arg;
+	dbi.dbi_fp = dba->dba_fp;
 	dbi.dbi_buf = buf;
 	dbi.dbi_len = len;
 
-#if defined(HAVE_LARGE_STACKS)
-	dump_bytes_cb(&dbi);
+#ifdef USE_SEND_TASKQ
+	taskqid_t id = taskq_dispatch(dba->dba_tq, dump_bytes_cb, &dbi,
+	    TQ_SLEEP);
+	if (id)
+		taskq_wait_id(dba->dba_tq, id);
```

Comment:
Its OK, but since I suppose there only one task at a time, I think we could use cheaper taskq_dispatch_ent() and taskq_wait().  I am not sure for what size of the buffer this is called for, but I suppose it may be not big.